ClientFactory/CurlFactory.php

```diff
@@ -40,6 +40,18 @@ public function createClient(array $config = [])
             throw new \LogicException('To use the Curl client you need to install the "php-http/curl-client" package.');
         }
 
+        // Try to resolve curl constant names
+        foreach ($config as $key => $value) {
+            // If the value starts with 'CURL' we assume it is a reference to a constant.
+            if (strpos($key, 'CURL') === 0) {
+                $contantValue = constant($key);
```

Comment:
Okey. Then I'm safe to assume that everything that comes in is either an Int or a name of a constant. If the constant is not defined I'll throw an exception. If the Int is invalid CURL will throw an exception